addons/account/models/account_partial_reconcile.py

```diff
@@ -546,8 +546,11 @@ def _create_tax_cash_basis_moves(self):
                     # that is actually paid by the current partial.
                     # ==========================================================================
 
-                    # Percentage expressed in the foreign currency.
-                    amount_currency = line.currency_id.round(line.amount_currency * partial_values['percentage'])
+                    if caba_treatment == 'tax' and move_values['is_fully_paid'] and line.amount_residual_currency:
+                        amount_currency = line.amount_residual_currency
+                    else:
+                        # Percentage expressed in the foreign currency.
+                        amount_currency = line.currency_id.round(line.amount_currency * partial_values['percentage'])
```

Comment:
I think there is a risk to exceed the residual amount when the invoice is not fully paid
```suggestion
                    # Percentage expressed in the foreign currency.
                    amount_currency = line.currency_id.round(line.amount_currency * partial_values['percentage'])
                    if caba_treatment == 'tax':
                        if move_values['is_fully_paid']:
                            amount_currency = line.amount_residual_currency
                        else:
                            sign = -1 if line.amount_currency < 0.0 else 1
                            amount_currency = sign * min(abs(amount_currency), abs(line.amount_residual_currency))
```